agent/agent.go

```diff
@@ -157,6 +157,25 @@ func startCommand(agentConfig config.AgentConfig, cmdArgs []string) (int, error)
 	return pid, err
 }
 
+func setupOpenfilesLimit() error {
+	var rlimit syscall.Rlimit
+	err := syscall.Getrlimit(syscall.RLIMIT_NOFILE, &rlimit)
+	if err != nil {
+		log.Errorf("Failed to get resource limits: %v", err)
+		return err
```

Comment:
Making this system call would recall Service Connect agent process to either run as a privileged user or have `cap_sys_resource` capability set. Would we get an appropriate error message here if that capability is not set? Otherwise, I think it's better to check that first and only make the syscall if allowed.

Reply:
Thanks for your advice. I agree that we should check the capability first before attempting to raise the limit.

Reply:
Hi, when increasing the soft limit of open files to match the hard limit for the current process, both non-root and root users can perform this operation without needing the CAP_SYS_RESOURCE capability. Considering this, I believe we can omit the check for the CAP_SYS_RESOURCE capability in the code. What do you think about this approach? 

> The hard  limit acts as a ceiling for the soft limit: an unprivileged process may set only its soft limit to a value in the range from 0 up to the hard limit, and  (irreversibly)  lower its  hard  limit.   A  privileged  process  (under  Linux:  one  with the CAP_SYS_RESOURCE capability) may make arbitrary changes to either limit value.

https://manpages.ubuntu.com/manpages/trusty/man2/setrlimit.2.html

Reply:
Thanks for the reference! Reading the documentation, I think it makes sense to not check for the capability since we're only increasing the soft limit to allowed max value of hard limit.

But I think we'll need to test whether or not the child process inherits this raised limit. Let me build a custom image with this commit and we can test that out.